profile_xf05id1/startup/11-secret_undulator.py

```diff
@@ -0,0 +1,281 @@
+from ophyd import (EpicsSignal, EpicsSignalRO, EpicsMotor,
+                   Device, Signal, PseudoPositioner, PseudoSingle)
+from ophyd.utils.epics_pvs import set_and_wait
+from ophyd.ophydobj import StatusBase, MoveStatus
+from ophyd import Component as Cpt
+
+from bluesky import Msg
+from bluesky.plan_tools import trigger_and_read, wrap_with_decorator, run_wrapper
+
+class UVDoneMOVN(PermissiveGetSignal):
+    def __init__(self, parent, moving, readback, actuate='actuate', stop='stop_signal',
+                 **kwargs):
+        super().__init__(parent=parent, value=1, **kwargs)
+        self._rbv = readback
+        self._brake = moving
+        self._act = actuate
+        self._stp = stop
+        self.target = None
+
+    def put(self, *arg, **kwargs):
+        raise TypeError("You con not tell an undulator motor it is done")
+
+    def _put(self, *args, **kwargs):
+        return super().put(*args, **kwargs)
+
+    def _watcher(self, obj=None, **kwargs):
+        target = self.target
+        if target is None:
+            return
+
+        rbv = getattr(self.parent, self._rbv)
+        stop = getattr(self.parent, self._stp)
+        moving = getattr(self.parent, self._brake)
+
+        cur_value = rbv.get()
+        not_moving = not moving.get()
+
+        # come back and check this threshold value
+        if not_moving and abs(target - cur_value) < 0.002:
+            self._put(1)
+            self._remove_cbs()
+            return
+
+        # if it is not moving, but we are not where we want to be, 
+        # poke it again
+        if not_moving:
+            actuate = getattr(self.parent, self._act)
+
+            print('re actuated')
+            actuate.put(1)
+
+    def _stop_watcher(self, *arg, **kwargs):
+        print('STOPPED')
+        self.reset(None)
+        stop = getattr(self.parent, self._stp)
+        self._put(1)
```

Comment:
marking is done is important so that BS will not hang is if waiting on this.

This needs to pick up the ability to mark 'fail' and convince the run engine to raise an exception in this case.